docs/source/cluster_management/sweep.rst

```diff
@@ -59,40 +59,33 @@ Note that some of these parameters are just used as a hint. Sweep dynamically mo
-In this setup, limiting ``candidateBatchHint`` to 100 would ensure that at most 10,000 cells are swept at a time, but would split the first table in 10 batches when only 1 was necessary.
-Instead, setting ``readLimit`` to 10,000 would allow both tables to be swept in optimal batches.
+``candidateBatchHint`` specifies the number of values to load from the KVS on each round-trip. ``readLimit`` specifies the number of values to read from the KVS in a batch.
+If each batch takes little time, it's advisable to increase the ``readLimit`` and ``deleteBatchHint``, to make sweep's batches bigger.
+If the client is OOMing, it's advisable to decrease the ``readLimit``, to have a lower number of values per batches.
+Since each batch contains at least ``candidateBatchHint`` values, if ``readLimit`` reaches the same value as ``candidateBatchHint`` you should reduce the latter config.
+Note that since sweep still needs to sweep at least a full row on every batch, it might be the case that the client is OOMing because the row its trying to sweep is very large. Please contact the AtlasDB team if you think you're hitting this issue.
 
-Memory pressure in Cassandra: decrease timestampsGetterBatchSize
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Memory pressure in Cassandra: decrease candidateBatchHint
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The sweep job works by requesting at least one full row at a time from the KVS, including all historical versions of each cell in the row.
-In some rare cases, the Cassandra KVS will not be able to construct a single full row in memory, likely due to specific cells being overwritten many times and/or containing very large values.
-This situation will manifest with Cassandra OOMing during sweep even if ``sweepBatchSize`` is set to 1.
-If that happens, you can use ``timestampsGetterBatchSize`` to instruct Cassandra to read a smaller number of columns at a time before aggregating them into a single row metadata to pass on to the sweeper.
-If ``timestampsGetterBatchSize`` is set, Cassandra will read at most one row at a time.
-The other batch parameters are still respected, but their values are unlikely to make a difference because the execution time will be dominated by the single-row reads.
+The sweep job works by requesting ``candidateBatchHint`` values from the KVS in each round-trip.
+In some rare cases, the Cassandra KVS will not be able to fetch ``candidateBatchHint`` values, likely due to specific cells being overwritten many times and/or containing very large values.
+If that happens, KVS calls will timeout and sweep will reduce ``candidateBatchHint`` automatically, trying to fetch less values in the following round-trips to the KVS.
+You can check the sweep logs to verify if this is happening frequently — and if this is the case — reduce this config to a value that sweep is able to run without failures.
```

Comment:
Could note here that, once it's successful for a while, sweep will try to increase the batch size back to the configured maximum, so if the maximum is truly too high (rather than a one off failure), this will happen frequently.